backend/scripts/update-dev-db.ts

```diff
@@ -0,0 +1,123 @@
+  const configAcc = sa;
+  let parsedPK;
+  try {
+    parsedPK = isProd
+      ? JSON.parse(process.env.FIREBASE_PRIVATE_KEY as string)
+      : JSON.parse(process.env.FIREBASE_DEV_PRIVATE_KEY as string);
+  } catch (err) {
+    parsedPK = isProd ? process.env.FIREBASE_PRIVATE_KEY : process.env.FIREBASE_DEV_PRIVATE_KEY;
+  }
+  configAcc.private_key = parsedPK;
+  configAcc.private_key_id = isProd
+    ? process.env.FIREBASE_PRIVATE_KEY_ID
+    : process.env.FIREBASE_DEV_PRIVATE_KEY_ID;
+  return configAcc;
+};
+
+async function deleteCollection(
+  db: FirebaseFirestore.Firestore,
+  collectionPath: string,
+  batchSize: number
+) {
+  const collectionRef = db.collection(collectionPath);
+  const query = collectionRef.orderBy('__name__').limit(batchSize);
+
+  return new Promise((resolve, reject) => {
+    deleteQueryBatch(db, query, resolve).catch(reject);
+  });
+}
+
+async function deleteQueryBatch(
+  db: FirebaseFirestore.Firestore,
+  query: FirebaseFirestore.Query,
+  resolve
+) {
+  const snapshot = await query.get();
+
+  const batchSize = snapshot.size;
+  if (batchSize === 0) {
+    // When there are no documents left, we are done
+    resolve();
+    return;
+  }
+
+  // Delete documents in a batch
+  const batch = db.batch();
+  snapshot.docs.forEach((doc) => {
+    batch.delete(doc.ref);
+  });
+  await batch.commit();
+
+  // Recurse on the next process tick, to avoid
+  // exploding the stack.
+  process.nextTick(() => {
+    deleteQueryBatch(db, query, resolve);
+  });
+}
+
+const readDbData = async (prodDb: FirebaseFirestore.Firestore): Promise<DbData> => {
+  const allCollections = await prodDb.listCollections();
+  return Promise.all(
+    allCollections.map(async (collection) => {
+      const docRefs = (await collection.get()).docs;
+      const docs = docRefs.map((docRef) => ({ id: docRef.id, data: docRef.data() }));
+      return {
+        id: collection.id,
+        docs
+      };
+    })
+  );
+};
+
+const rewriteDbData = async (devDb: FirebaseFirestore.Firestore, data: DbData): Promise<void[]> => {
+  const allCollections = await devDb
+    .listCollections()
+    .then((collections) => collections.map((collection) => collection.id));
+
+  await Promise.all(allCollections.map((collection) => deleteCollection(devDb, collection, 10)));
+  return data.map((collection) =>
+    collection.docs.forEach((doc) => devDb.doc(`${collection.id}/${doc.id}`).set(doc.data))
+  );
+};
+
+const main = async () => {
+  const prodApp = admin.initializeApp(
+    {
+      credential: admin.credential.cert(configureAccount(prodServiceAccount, true)),
+      databaseURL: 'https://idol-b6c68.firebaseio.com',
+      storageBucket: 'gs://cornelldti-idol.appspot.com'
+    },
+    'prod'
+  );
+  const devApp = admin.initializeApp(
+    {
+      credential: admin.credential.cert(configureAccount(devServiceAccount, false)),
+      databaseURL: 'https://idol-b6c68.firebaseio.com',
```

Comment:
Is this url supposed to be different from the one for the prod db? Looking at the firebase console, the dev db has a different url, but it seems to be a weird default one.

Reply:
Tbh I'm not even 100% if it needs to be included. I think this db URL is referring to firebase's realtime database which is different than the firestore database. We don't even ever use this feature of firebase.